test/e2e/scenarios/remove_cluster.go

```diff
@@ -0,0 +1,23 @@
+package scenarios
+
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = PDescribe("[Ctlplane] Removing policies related to removed cluster", func() {
+	PIt("Should remove policies relevant to removed cluster", func() {
+
+		By("creating listener pod 1 with label 1 in cluster 1 in namespace 1")
+		By("creating listener pod 2 with label 2 in cluster 2 in namespace 1")
+		By("Registering cluster 3 to kubefed")
+		By("creating listener pod 3 with label 3 in cluster 3 in namespace 3")
+		By("Adding network policy to cluster1 to allow connection from pods in cluster 2 and 3")
+
+		By("Waiting for a NetworkPolicy to appear in cluster 1 containing pod 3 IP in ipBlocks")
+
+		By("Unregistering cluster 3")
+		By("Discover the removed cluster")
```

Comment:
I think you don't need to discover removed cluster here, that will be done by coastguard.